observability/logging/logging-6.0/log6x-clf.adoc

```diff
@@ -0,0 +1,91 @@
+:_mod-docs-content-type: ASSEMBLY
+include::_attributes/common-attributes.adoc[]
+[id="log6x-CLF_{context}"]
+= Configuring log forwarding
+:context: logging-6x
+
+toc::[]
+
+The `ClusterLogForwarder` CR serves as a single point of configuration for log forwarding, making it easier to manage and maintain log collection and forwarding rules.
+
+* Defines inputs (sources) for log collection
+* Specifies outputs (destinations) for log forwarding
+* Configures filters for log processing
+* Defines pipelines to route logs from inputs to outputs
+* Indicates management state (managed or unmanaged)
+
+// Needs engineering eval re applicability to 6.0
+include::modules/log-forwarding-implementations.adoc[leveloffset=+1]
+
+// This needs to be called out extensively.
+[source,yaml]
+----
+apiVersion: observability.openshift.io/v1
+kind: ClusterLogForwarder
+metadata:
+  name: instance
+spec:
+  inputs:
+    - application
+    - infrastructure
+    - audit
+  outputs:
+    - lokiStack
+    - elasticsearch
+    - splunk
+  filters:
+    - kubeAPIAudit
+    - detectMultilineException
+    - parse
+    - openshiftLabels
+    - drop
+    - prune
+  pipelines:
+    - name: application-logs
+      input: application
+      output: lokiStack
+      filters:
+        - parse
+        - openshiftLabels
+  managementState:
+    managed: true
+----
+
+[id="log6x-CLF-tuning_{context}"]
+== Output tuning
+//need to validate how much of this applies to 6.0 & edit as needed.
+include::modules/logging-delivery-tuning.adoc[leveloffset=+1]
+
+
+[id="log6x-CLF-filters_{context}"]
+== Filtering logs
+
+The following filters are available:
+// Should be an include for each, create: modules/log6<filtername>.adoc - Associated JIRA?
+detectMultilineException:: Detects and handles multiline exceptions.
+parse:: Parses log messages using Vector's parsing language.
+openshiftLabels:: Adds OpenShift-specific labels to log messages.
+
+// All of these need to be validated by engineering for 6.0.
+include::modules/logging-audit-log-filtering.adoc[leveloffset=+1]
+
+include::modules/logging-content-filter-drop-records.adoc[leveloffset=+1]
+
+include::modules/logging-content-filter-prune-records.adoc[leveloffset=+1]
+
+include::modules/logging-input-spec-filter-audit-infrastructure.adoc[leveloffset=+1]
+
+include::modules/logging-input-spec-filter-labels-expressions.adoc[leveloffset=+1]
+
+include::modules/logging-input-spec-filter-namespace-container.adoc[leveloffset=+1]
+
+include::modules/logging-multiline-except.adoc[leveloffset=+1]
+
+
+[id="log6x-CLF-samples_{context}"]
+== Use case samples
+// Should be an include, create: modules/log6x-use-multi-instance.adoc - Associated JIRA?
+* Multiple-Instance Mode
+
+// Should be an include, create: modules/log6x-use-metric-export.adoc - Associated JIRA?
+* Log File Metric Exporter as a Separate Deployment
```

Comment:
This is still a feature. If users wish to see the metrics produced by this component then they will need to deploy it